optuna/_optimize.py

```diff
@@ -193,6 +193,8 @@ def _run_trial(
     func_err_fail_exc_info: Optional[Any] = None
     # Set to a string if `func` returns correctly but the return value violates assumptions.
     values_conversion_failure_message: Optional[str] = None
+    # update the trial state to running and **change the datetime_start**
+    study._storage.set_trial_state(trial._trial_id, TrialState.RUNNING)
```

Comment:
Might be better to do this inside `Study.ask`. With this logic, the timestamp will not be updated when the users uses `Study.ask` instead of `Study.optimize`. You might have to update the base branch to the latest master since this method was introduced very recently.

Reply:
You could perhaps catch up on the ask and tell API from this issue https://github.com/optuna/optuna/issues/2177, but basically, it's an alternative to `Study.optimize` where the user directly creates and manages trial lifetimes.